diff --git a/node_modules/@nightingale-elements/nightingale-msa/src/nightingale-msa.ts b/node_modules/@nightingale-elements/nightingale-msa/src/nightingale-msa.ts
index a2fb149..3f9b144 100644
--- a/node_modules/@nightingale-elements/nightingale-msa/src/nightingale-msa.ts
+++ b/node_modules/@nightingale-elements/nightingale-msa/src/nightingale-msa.ts
@@ -14,6 +14,7 @@ import NightingaleElement, {
 } from "@nightingale-elements/nightingale-new-core";
 import object2style from "./utils/object2style";
 import { Region, SequencesMSA } from "./types/types";
+import ConservationWorker from "./workers/conservation.worker.ts";
 
 const DEAFULT_TILE_HEIGHT = 20;
 const DEAFULT_COLOR_SCHEME = "clustal2";
@@ -53,9 +54,7 @@ class NightingaleMSA extends withManager(
   })
   overlayConservtion?: boolean = false;
 
-  worker = new Worker(
-    new URL("./workers/conservation.worker.ts", import.meta.url)
-  );
+  worker= new ConservationWorker();
   
   private sequenceViewer?: SequenceViewerComponent | null;
   private labelPanel?: LabelsComponent | null;
